internal/grpcutil/target.go

```diff
@@ -60,9 +61,45 @@ func ParseTarget(target string, skipUnixColonParsing bool) (ret resolver.Target)
 		return resolver.Target{Endpoint: target}
 	}
 	if ret.Scheme == "unix" {
-		// Add the "/" back in the unix case, so the unix resolver receives the
-		// actual endpoint.
-		ret.Endpoint = "/" + ret.Endpoint
+		if !skipUnixColonParsing {
+			// Add the "/" back in the unix case, so the unix resolver receives the
+			// actual endpoint.
+			ret.Endpoint = "/" + ret.Endpoint
+		} else {
+			// Custom dialer should receive "unix:///[...]".
+			return resolver.Target{Endpoint: target}
```

Comment:
This should not be here; instead make the transport invoke the user's dialer after prepending "unix://" to the address string.

Reply:
Removed that part and am now appending to the address in transport.